src/test/java/com/basho/riak/client/api/commands/itest/ITestGetCoverage.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertTrue;
+
+public class ITestGetCoverage extends ITestBase {
+    private static Logger logger = LoggerFactory.getLogger(ITestGetCoverage.class);
+
+    @Test(timeout = 60000)
+    public void getFullCoverage() throws ExecutionException, InterruptedException, UnknownHostException {
+        logger.info("Retrieve list of EPs...");
+        final RiakClient client = new RiakClient(cluster);
+        final GetCoverage cmd = new GetCoverage.Builder()
+                .withForcedUpdate()
+                .build();
+
+        final GetCoverage.Response response = client.execute(cmd);
+
+        logger.info("Got the following list of EPs: {}", response.entryPoints());
+
+        /**
+         * Since GetCoverage command is the only way to get an information about the number of available Riak nodes
+         * there is no chance to verify count of returned EPs. Therefore we will iterate through the list of the
+         * returned EPs and ping each of them to ensure that at least a valid EP was returned.
+         * */
+        verifyThatAllNodesAreAvailable(response.entryPoints());
+    }
+
+    @Test(timeout = 60000)
+    public void getCoverageForLocation() throws ExecutionException, InterruptedException, UnknownHostException {
+        final RiakClient client = new RiakClient(cluster);
+        final String key = "coverage-4-location";
+
+        // create test value
+        final RiakObject ro = new RiakObject()
+                .setValue(BinaryValue.create("{\"value\": 42}"))
+                .setContentType("application/json");
+
+        final StoreValue request = new StoreValue.Builder(ro)
+                .withLocation(new Location(defaultNamespace(), key))
+                .build();
+
+        client.execute(request);
+
+        // perform test
+        final Location location = new Location(defaultNamespace(), key);
+        logger.info("Retrieve list of EPs for location '{}' ...", location);
+
+        final GetCoverage cmd = new GetCoverage.Builder()
+                .withForcedUpdate()
+                .withLocation(location)
+                .build();
+
+        final GetCoverage.Response response = client.execute(cmd);
+        logger.info("Got the following list of EPs: {}", response.entryPoints());
+
+        /**
+         * Since GetCoverage command is the only way to get an information about the number of available Riak nodes
+         * there is no chance to verify count of returned EPs. Therefore we will iterate through the list of the
+         * returned EPs and ping each of them to ensure that at least a valid EP was returned.
+         * */
+        verifyThatAllNodesAreAvailable(response.entryPoints());
+    }
+
+    private static void verifyThatAllNodesAreAvailable(Collection<Map.Entry<String,Integer>> nodes) throws UnknownHostException, InterruptedException {
+        for(Map.Entry<String,Integer> e: nodes){
```

Comment:
@srgg In fact, we 'respectfully disagreed' on the matter, but I have since become convinced that not_routed makes little sense without indication on which node the failing listener is. I have made changes to filter such non-addresses in the returned results, and updated my PRs in riak_api and riak_kv.